common/src/main/java/io/novafoundation/nova/common/validation/InputValidationMixinFactory.kt

```diff
@@ -0,0 +1,25 @@
+package io.novafoundation.nova.common.validation
+
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.map
+
+class InputValidationMixinFactory<P, E>(
+    private val validationSystem: ValidationSystem<P, E>,
+) {
+    fun create(inputFlow: Flow<String>, payloadFormatter: (String) -> P): InputValidationMixin<P, E> {
+        return InputValidationMixin(validationSystem, inputFlow, payloadFormatter)
+    }
+}
+
+class InputValidationMixin<P, E>(
+    private val validationSystem: ValidationSystem<P, E>,
+    private val inputFlow: Flow<String>,
+    private val payloadFormatter: (String) -> P
```

Comment:
We can pass `Flow<P>` to be more generic and let the factory handle the transformation